islandora_audio.module

```diff
@@ -210,6 +210,19 @@ function islandora_audio_islandora_sp_audiocmodel_islandora_ingest_steps() {
  * Implements hook_CMODEL_PID_islandora_derivative().
  */
 function islandora_audio_islandora_sp_audiocmodel_islandora_derivative() {
+  if (variable_get('islandora_audio_defer_derivatives_on_ingest', FALSE)) {
+    return array(
+      array(
+        'source_dsid' => NULL,
+        'destination_dsid' => 'TN',
```

Comment:
I don't really understand the question.  Let me go over why I was creating the code change.

The problem that I needed to solve was to not install any additional executables on the Drupal server, so I could not install lame.  In order to use the JW player with the audio module, the Fedora object needs an MP3 in the PROXY_MP3 datastream.  So, the team I am working with has the PROXY_MP3 datastream added on the Fedora server.  This worked by leaving an invalid path to the lame encoder in the audio solution pack administration.  However, simply leaving an invalid path to the lame encoder path did 2 things which the server administrator did not like: needing to use an override when running update.php, and clutter the log each time an audio file was ingested.  

The intent of the change was to remove those two issues for the adminstrator.

The addition of the TN datastream does not require any special libraries because the datastream content is a file within the audio module's images directory.  In no way is it actually a derivative of the audio file.  The code in islandora_audio_islandora_sp_audiocmodel_islandora_derivative is a simple way to hook into adding a datastream during ingest.  This methodology was already present in a few lines beneath the quoted section.